### ISSUE TITLE
Plugin Update Command Not Listing All Plugins

### ISSUE THREAD

**Issue by srguglielmo** [wp-cli/extension-command#63]
Hello,

I have a large WordPress multisite. I use wp cli to manage WordPress core, plugins, themes, sites, and users. Recently, I updated wp-cli to version 1.4.0. It seems that plugins are now not missing from the `wp plugin` command. For example, when I log into WordPress via a browser and navigate to `/wp-admin/network/plugins.php` to list all plugins on the network, I see the following:

![Screenshot of WP Plugins](https://i.imgur.com/y2ebJt0.png)

I have WordFence installed and see the following plugins that have updates available:

![Screenshot of WordFence Plugin Updates](https://i.imgur.com/e5ywWOz.png)

However, when I execute `wp plugin update --all --dry-run`, I get the following output. As you can see, many of the plugins are missing from this list, despite the fact that I included `--all`.

```
Available plugin updates:
+----------------------+----------+-----------+----------------+
| name                 | status   | version   | update_version |
+----------------------+----------+-----------+----------------+
| amr-ical-events-list | inactive | 5.1       | 5.3            |
| feedwordpress        | inactive | 2017.1004 | 2017.1020      |
| jetpack              | active   | 5.4       | 5.5            |
| the-events-calendar  | inactive | 4.6.1     | 4.6.4          |
| wordpress-seo        | inactive | 5.6.1     | 5.7.1          |
+----------------------+----------+-----------+----------------+
```

This, to me, is a bug. I actually did not realize I was missing all these plugin updates because I do not usually check the website (I use wp-cli).

I do see that in recent commits, the `wp plugin` command now uses some type of WordPress core filter to selectively show plugins. However, I do not see a way to bypass this. When I log into the site, I see all plugins listed with their updates, but I do not see them in wp-cli.

Thanks,
Steve

**Comment by danielbachhuber** [comment]
Hi @srguglielmo,

Does the issue reproduce when you skip WordFence with `--skip-plugins=wordfence`? Also, can you use the `--skip-plugins=<plugin-name>` flag to debug whether the issue is caused by a specific plugin?

**Comment by srguglielmo** [comment]
Hi,

Skipping wordfence does not help:

```
$ wp --skip-plugins=wordfence plugin update --all --dry-run
Available plugin updates:
+----------------------+----------+-----------+----------------+
| name                 | status   | version   | update_version |
+----------------------+----------+-----------+----------------+
| amr-ical-events-list | inactive | 5.1       | 5.3            |
| feedwordpress        | inactive | 2017.1004 | 2017.1020      |
| jetpack              | active   | 5.4       | 5.5            |
| the-events-calendar  | inactive | 4.6.1     | 4.6.4          |
| wordpress-seo        | inactive | 5.6.1     | 5.7.1          |
+----------------------+----------+-----------+----------------+
```

However, after testing various `--skip-plugin` values, I discovered that it works correctly with just `--skip-plugins` (with no value; presumably it skips **all** plugins).

```
$ wp --skip-plugins --skip-themes plugin update --all --dry-run
Available plugin updates:
+--------------------------+----------------+-----------+----------------+
| name                     | status         | version   | update_version |
+--------------------------+----------------+-----------+----------------+
| akismet                  | inactive       | 3.3.2     | 4.0.1          |
| amr-ical-events-list     | inactive       | 5.1       | 5.3            |
| attachments              | inactive       | 3.5.7     | 3.5.9          |
| collapsing-categories    | inactive       | 2.0.8     | 2.1            |
| email-subscribers        | inactive       | 3.3.3     | 3.4.2          |
| feedwordpress            | inactive       | 2017.1004 | 2017.1020      |
| gd-bbpress-attachments   | inactive       | 2.4       | 2.5            |
| image-widget             | active-network | 4.4.5     | 4.4.6          |
| jetpack                  | active         | 5.4       | 5.5            |
| members                  | inactive       | 1.1.3     | 2.0.2          |
| more-privacy-options     | active-network | 4.3       | 4.6            |
| superb-slideshow-gallery | inactive       | 12.2      | 12.3           |
| the-events-calendar      | inactive       | 4.6.1     | 4.6.4          |
| user-switching           | active-network | 1.0.9     | 1.3.0          |
| wordfence                | active-network | 6.3.12    | 6.3.21         |
| wordpress-seo            | inactive       | 5.6.1     | 5.7.1          |
+--------------------------+----------------+-----------+----------------+
```

I suppose there must be a plugin installed that is somehow filtering...the list of plugins? It's odd because I can login and see all plugins listed on the website with no problem.

**Comment by danielbachhuber** [comment]
> I suppose there must be a plugin installed that is somehow filtering...the list of plugins?

Most likely. It's hard to say without having access to debug further. Happy to help out in Slack.

**Comment by amandafrench** [comment]
I recently started having this issue too on a large multisite network. 'wp list plugins' would only list must-use and drop-in plugins, plus a few network-activated ones. The --skip-plugins trick mentioned here worked for me, as well, and allowed me to update "missing" plugins. 

I deactivated all the plugins on the network and started reactivating them one by one to see which is the culprit. In my case it's a useful but old multisite plugin called (appropriately) "exclude-plugins" that allows me to display only certain plugins to subsite admins. https://wordpress.org/plugins/exclude-plugins/ When I run 'wp plugin list' or 'wp plugin update' and add the flag --skip-plugins=exclude-plugins, the commands work as expected.
